clarity/contracts/arkadiko-auction-engine-v4-1.clar

```diff
@@ -0,0 +1,295 @@
+        total-debt-burned: u0,
+        total-collateral-sold: u0,
+        ended-at: u0,
+      })
+    )
+      (asserts! (>= liquidation-ratio collateral-to-debt-ratio) (err ERR-NO-LIQUIDATION-REQUIRED))
+
+      ;; Add auction
+      (map-set auctions { id: auction-id } auction)
+
+      ;; Try to burn
+      (try! (burn-usda auction-id oracle coll-type vault-manager ft reserve liquidation-pool liquidation-rewards))
+
+      (print { type: "auction", action: "created", data: auction })
+      (ok true)
+    )
+  )
+)
+
+;; @desc burn USDA for vault and get part of collateral
+;; @param auction-id; the auction
+;; @param oracle; oracle
+;; @param coll-type; collateral types
+;; @param vault-manager; vault manager
+;; @param ft; collateral token if not STX vault
+;; @param reserve; reserve to get collateral tokens from
+;; @param liquidation-pool; pool to get USDA from
+;; @param liquidation-rewards; contract to deposit rewards
+(define-public (burn-usda
+  (auction-id uint)
+  (oracle <oracle-trait>)
+  (coll-type <collateral-types-trait>)
+  (vault-manager <vault-manager-trait>)
+  (ft <ft-trait>)
+  (reserve <vault-trait>)
+  (liquidation-pool <liquidation-pool-trait>)
+  (liquidation-rewards <liquidation-rewards-trait>)
+)
+  (let (
+    (auction (get-auction-by-id auction-id))
+    (debt-left (- (get debt-to-raise auction) (get total-debt-burned auction)))
+    (usda-to-use (withdrawable-usda debt-left liquidation-pool))
+    (collateral-sold (unwrap-panic (get-collateral-amount oracle auction-id usda-to-use)))
+
+    (token-string (if (is-eq (get collateral-token auction) "STX") "xSTX" (get collateral-token auction)))
+    (token-address (get collateral-address auction))
+  )
+    (asserts! (not (shutdown-activated)) (err ERR-EMERGENCY-SHUTDOWN-ACTIVATED))
+    (asserts! (is-eq (contract-of vault-manager) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "freddie"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of coll-type) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "collateral-types"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of oracle) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "oracle"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of reserve) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "sip10-reserve"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of liquidation-pool) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "liquidation-pool"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of liquidation-rewards) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "liquidation-rewards"))) (err ERR-NOT-AUTHORIZED))
+
+    (asserts! (or (is-eq (contract-of ft) token-address) (is-eq "xSTX" token-string)) (err ERR-TOKEN-TYPE-MISMATCH))
```

Comment:
Q: why do you need to xSTX comparison? xSTX is also a SIP10 FT that has a `token-address`, no? In that case, we would expect the `ft` argument to be `.wrapped-stx-token`?